config/opal_check_pmi.m4

```diff
@@ -39,18 +39,30 @@ AC_DEFUN([OPAL_CHECK_PMI_LIB],
     opal_check_$3_mycppflags=
 
     # check for the header
-    AC_MSG_CHECKING([for $3.h in $1/include])
-    AS_IF([test -f $1/include/$3.h],
-          [AC_MSG_RESULT([found])
-           opal_check_$3_mycppflags="-I$1/include"],
-          [AC_MSG_RESULT([not found])
-           AC_MSG_CHECKING([for $3.h in $1/include/slurm])
-           AS_IF([test -f $1/include/slurm/$3.h],
+    AS_IF([test -n "$1"],
+          [AC_MSG_CHECKING([for $3.h in $1/include])
```

Comment:
I note that we never just check to see if the header is present in where they told us to look - everything immediately puts an ```/include``` after it. I believe people have been complaining about that - shouldn't this check pattern start by checking where they told us it is?

Reply:
well, the goal of this PR is to stop ending up with `-L/usr/lib[64]` in the `LDFLAGS`, and I kept everything else unmodified.
That being said, this is a fair point and I will address this in a new commit and append it to this PR and its siblings for the other release branches.
Shall I stop searching in `$1/include/slurm` when `$1` is not empty then ?

Reply:
I would continue that search as slurm does that strange thing, and users have come to expect it.